### PR TITLE
added --enabled and --disabled options to occ app:list

### LINKED ISSUE #32699: occ app:list --disabled
Above command can help admins to automate things.
Right now the disabled apps are shown as a subset of "app:list", but not in isolation.

### PR DESCRIPTION
**Pull request by adam-blakey**
This deals with issue #32699.

@tobiasKaminsky, @PVince81

### CODE REVIEW

**Comment by PVince81** [comment]
- [ ] BUG: running `occ app:list` alone shows an error but should actually show both enabled and disabled.

I think you can remove the error when both are specified.

**Comment by PVince81** [comment]
seems it doesn't work:
image

can you recheck ?

File: core/Command/App/ListApps.php [hunk 1]
```diff
@@ -102,11 +141,15 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 	protected function writeAppList(InputInterface $input, OutputInterface $output, $items) {
 		switch ($input->getOption('output')) {
 			case self::OUTPUT_FORMAT_PLAIN:
-				$output->writeln('Enabled:');
-				parent::writeArrayInOutputFormat($input, $output, $items['enabled']);
-
-				$output->writeln('Disabled:');
-				parent::writeArrayInOutputFormat($input, $output, $items['disabled']);
+				if ($items['enabled']) {
```
Comment (PVince81) [inline core/Command/App/ListApps.php:144]: I think you need to call `isset()` here to avoid PHP errors.

same below

**Comment by JustAdamHere** [comment]
@PVince81 apologies, I'm an idiot. I've made those corrections and believe this may be working now. Could you check and let me know if this is now okay?

**Comment by JustAdamHere** [comment]
@PVince81 I somehow didn't get the notifications for the above mentions, so sorry for the delay. Could you take another look at this when you get the chance? Perhaps I'm being really silly but I can't figure out why the code is failing 4 of the above automatic tests. Any help would be appreciated.

**Review by skjnldsv — APPROVED** [review]

File: core/Command/App/ListApps.php [hunk 2]
```diff
@@ -61,6 +74,23 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
+			$showDisabledApps = false;
+		}
+		else if ($input->getOption('disabled')) {
+			$showEnabledApps = false;
+			$showDisabledApps = true;
+		}
+		else {
+			$showEnabledApps = true;
+			$showDisabledApps = true;
+		}
```
Comment (come-nc) [inline core/Command/App/ListApps.php:92]:
```suggestion
		$showEnabledApps = $input->getOption('enabled') || !$input->getOption('disabled');
		$showDisabledApps = $input->getOption('disabled') || !$input->getOption('enabled');
```

**Review by come-nc — COMMENTED** [review]
All good apart from overcomplicated if.

**Review by come-nc — APPROVED** [review]
Tested and it works :+1:

**Comment by Altahrim** [comment]
@adam-blakey Could you rebase, reword your commit using conventional commit and fix the style please?
For style, you can use `composer cs:fix`

**Comment by nickvergessen** [comment]
> Performance testing / performance-8.0 (pull_request) Failing after 9m 

is expected for forks

> Block unconventional commits / Block unconventional commits (pull_request) 

is fine for community, feel free to read https://www.conventionalcommits.org/en/v1.0.0/ and apply it next time :)

**Pull request merged**